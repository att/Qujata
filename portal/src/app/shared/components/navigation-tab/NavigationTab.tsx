import { NavLink, useLocation } from 'react-router-dom';
import styles from './NavigationTab.module.scss';
import cn from 'classnames';

export interface INavigationTab {
  title: string;
  link: string;
}
export interface NavigationTabProps {
  tabs: INavigationTab[];
}

export const NavigationTab: React.FC<NavigationTabProps> = (props: NavigationTabProps) => {
    const { tabs } = props;
    const location = useLocation();

    return (
        <div className={styles.tabs}>
           {tabs.map(((tab: INavigationTab, index: number) => (
            <div key={index}>
                <NavLink 
                    className={cn(styles.tab, {[styles.activeTab]: location.pathname === tab.link})} 
                    to={tab.link}             
                >
                    {tab.title}
                </NavLink>
            </div>
          )))}
        </div>
    );
}
