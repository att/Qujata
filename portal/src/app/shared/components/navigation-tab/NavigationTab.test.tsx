import { render, RenderResult } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { NavigationTab } from './NavigationTab';

const tabs = [
  {
    link: '/',
    title: 'Home',
  },
  {
    link: '/all-experiments',
    title: 'All Experiments',
  }
];
describe('NavigationTab', () => {
  test('renders Navigation Tab', () => {
    const { container }: RenderResult = render(<NavigationTab tabs={tabs} />, { wrapper: MemoryRouter });
    expect(container.firstChild).toMatchSnapshot();
  });
});
