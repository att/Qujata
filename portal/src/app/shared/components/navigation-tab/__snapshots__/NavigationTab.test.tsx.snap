// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationTab renders Navigation Tab 1`] = `
<div
  class="tabs"
>
  <div>
    <a
      aria-current="page"
      class="tab activeTab active"
      href="/"
    >
      Home
    </a>
  </div>
  <div>
    <a
      class="tab"
      href="/all-experiments"
    >
      All Experiments
    </a>
  </div>
</div>
`;
